### PR TITLE
Fuzzy skin fixes + on/off for first layer

### LINKED ISSUE #2891: Fuzzy skin deviates way too much from expected geometry
### OrcaSlicer Version

1.7.0

### OS version

Archlinux

### Additional system information

Not relevant

### Printer

Not relevant

### How to reproduce

Enable Fuzzy skin Contour with 
1 (point distance) and 0.2 (thickness)
Use some geometry that has relatively sharp corners.

### Actual results

before

### Expected results

after
This screenshot is taken from a patched version of OrcaSlicer. I looked at the fuzzy skin code and compared to the Cura implementation and I think I identified and fixed the issue. Would love to submit a PR but I am no slicer expert + this would be my first PR in Orca, so some help would be greatly appreciated.

### Project file & Debug log uploads

[bugreport.zip](https://github.com/SoftFever/OrcaSlicer/files/13466261/bugreport.zip)

### Checklist of files to include

- [ ] Log file
- [ ] Project file

### PR DESCRIPTION
**Pull request by andrewboktor**
Fixing fuzzy skin so it doesn't exit the desired geometry at line ends
Adding option to enable/disable fuzzy skin on first layer

Fixes #2891

### CODE REVIEW

**Review by Noisyfox — CHANGES_REQUESTED** [review]

File: src/libslic3r/PerimeterGenerator.cpp [hunk 1]
```diff
@@ -68,16 +68,14 @@ static void fuzzy_polygon(Polygon &poly, double fuzzy_skin_thickness, double fuz
     { // 'a' is the (next) new point between p0 and p1
         Vec2d  p0p1      = (p1 - *p0).cast<double>();
         double p0p1_size = p0p1.norm();
-        // so that p0p1_size - dist_last_point evaulates to dist_left_over - p0p1_size
-        double dist_last_point = dist_left_over + p0p1_size * 2.;
-        for (double p0pa_dist = dist_left_over; p0pa_dist < p0p1_size;
+        double p0pa_dist = dist_left_over;
+        for (; p0pa_dist < p0p1_size;
```
[the remark this answers is not in the document]
Comment (andrewboktor) [reply src/libslic3r/PerimeterGenerator.cpp:72]: This seems to deal with the situation where the line is too small to place a "pa" point on, in which case Cura seems to be placing a "pa" point in the middle of said line.
I feel like here, dist_left_over should deal with that (by accumulating length of short lines until we're eligible to add a "pa" point).
I will take a more in depth look later today. If you have geometry that makes this misbehave, please share, I think that would be really helpful.

[the remark this answers is not in the document]
Comment (Noisyfox) [reply src/libslic3r/PerimeterGenerator.cpp:72]: I don't have much experience with this feature TBH so I couldn't give any feedback on this. I think it shouldn't matter too much for most of the models anyway.
I'll approve the PR and if you think it's OK to merge without those lines then just let me know.

**Review by Noisyfox — APPROVED** [review]

File: src/libslic3r/PerimeterGenerator.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewboktor) [reply src/libslic3r/PerimeterGenerator.cpp:72]: @Noisyfox Those lines are not here anyway in the current state. So I think we should be fine merging as is. 
I've been slicing 4-5 different models and printing them perfectly with this PR.
If we have a resident expert on fuzzy skin, we should have them take a look, if not then we should be fine to merge.

[the remark this answers is not in the document]
Comment (Noisyfox) [reply src/libslic3r/PerimeterGenerator.cpp:72]: > Those lines are not here anyway in the current state. So I think we should be fine merging as is.

Fare enough, and we can always add those later if needed easily. Will merge as is, thanks!

**Pull request merged**